src/utils/utils.js

```diff
@@ -4,6 +4,19 @@ async function syncForEach(array, callback) {
   }
 }
 
+function checkHTTPAllowance(urls) {
+  urls.forEach(url => {
+    if (!/^https.*/.test(url)) {
+      if (process.env.ALLOW_HTTP !== 'yes') {
```

Comment:
I don't think that the method should be responsible of checking the environment variable (nor throwing, for that matter). I'd move most of this logic to the caller and leave just a `isHTTPS(url)` function here.

Reply:
After debating about the implementation and implications. `checkHTTPAllowance` was renamed to `checkHTTPS`, and changed its implementation to receive the `env` variable as a parameter, as well as it checks for only one URL at a time.